### ISSUE TITLE
Valkyrize calls to ActiveFedora::Base.find (removes 27 refs to ActiveFedora)

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#3790]
### Descriptive summary

Replace all calls to ActiveFedora::Base.find to use valkyrie query_service to find the object (except calls made in the lib/wings directories).  At this writing there are 27 uses of the ActiveFedora::Base #find method.

### ActiveFedora code

Typical call looks like...

```
ActiveFedora::Base.find(id)
```

search for `ActiveFedora::Base.find` to find all occurrences.

### Valkyrie replacement code could look something like...

Create `app/queries/hyrax/find_query.rb`.

Assumes id is the active_fedora id
```
      resource = query_service.find_by(id: Valkyrie::ID.new(id.to_s))
      af_object = Wings::ActiveFedoraConverter.new(resource: resource).convert # TODO: WINGS - Remove when Valkyrie fully integrated.
```

_NOTE: There is one instance that uses ActiveFedora::Base.find with two arguments, the second of which is cast.  This does not address that usage._

_NOTE: There is at least one instance where an array of ids is passed in instead of a single id._

Reference: Replaces https://github.com/samvera/active_fedora/blob/master/lib/active_fedora/relation/finder_methods.rb#L28-L49

### Considerations

* Wings::Valkyrie::QueryService #find_by looks for the alternate_identifier which is the ActiveFedora object's ID
* Since we are still working with ActiveFedora objects in the rest of the code, the Valkyrie resource has to be converted back to an ActiveFedora object.
* There is a Wings::IdConverterService that gets the Valkyrie id by using ActiveFedora to find the AF object and then getting the valkyrie id from it.  Can't use it in this case because it is a circular call since it uses the ActiveFedora::Base.find method we are trying to remove.  
* Do NOT replace this call if found in lib/wings directories.

### Impact on Testing

* Not expected to require changes to existing tests.  All tests should pass with the replacement code.

**Comment by elrayle** [comment]
There was discussion in slack this morning.  I believe we have landed on an approach that will ultimately end with native Valkyrie code in each location where `find` is being replaced.

The summary of that discussion follows.

Update query_service.find_by to allow it to return a resource or an AF object.  By default, it will return a resource.

```
QueryService.find_by(id: id, use_valkyrie: true)
```

Then all instances calling `find` can be changed to…

```
query_service.find_by(id: id, use_valkyrie: false)
```

As the code that calls `find_by` is converted to be able to use the returned resource instead of an af object, then the call will become…

```
query_service.find_by(id: id)
```

Once all uses of `query_service.find_by(id: id, use_valkyrie: false)` are changed to `query_service.find_by(id: id)`, then the extra `use_valkyrie` parameter can be removed from the query service's `find_by` method.

**Comment by elrayle** [comment]
All but 4 calls to ActiveFedora::Base.find were addressed by PR #3828 Replace ActiveFedora::Base.find #3828. 

The remaining calls are all in the nested indexing code.  See Issue #3829 Replace ActiveFedora usage in NestingIndexAdapter
